### PR TITLE
Catching File Exceptions in openpower-vpd-parser

### PR DESCRIPTION
**Pull request by jinuthomas**
In this commit, I have added code to handle file exceptions more effectively. By implementing proper exception handling, we can improve the robustness and reliability of the file operations within our codebase.

Here are the key changes made in this commit:

  - Introduced a try-catch block around the file operation sections.
  - Within the try block, added code to perform the necessary file operations.
  - Implemented catch blocks to handle specific file exceptions.
  - In each catch block, included appropriate error handling logic, such as logging the error message or displaying a user-friendly error message.
  - Ensured that the catch blocks gracefully handle the exceptions and prevent the program from crashing or behaving unexpectedly.
  - Fixed shellcheck issues with the wait-vpd-parsers.sh shell script (no functional change or relation to this commit)

By adding this exception handling code, we can anticipate and handle potential file-related errors gracefully, providing a smoother experience for users and preventing any unexpected crashes or data loss. This would also aid in debugging issues.

### CODE REVIEW

**Comment by jinuthomas** [comment]
jenkins run tests please

File: impl.hpp [hunk 1]
```diff
@@ -78,14 +78,17 @@ class Impl
...
         {
             vpdFileStream.open(vpdFilePath,
                                std::ios::in | std::ios::out | std::ios::binary);
         }
-        catch (const std::fstream::failure& e)
+        catch (const std::fstream::failure& fail)
         {
-            std::cout << e.what();
+            std::cerr << "Exception in file handling [" << vpdFilePath
+                      << "] error : " << fail.what();
```
[the remark this answers is not in the document]
Comment (jinuthomas) [reply impl.hpp:90]: the std::err has the description so did not add error();

File: vpd_tool_impl.cpp [hunk 2]
```diff
@@ -650,13 +650,24 @@ void VpdTool::readKwFromHw(const uint32_t& startOffset)
...
     completeVPDFile.resize(65504);
     fstream vpdFileStream;
-    vpdFileStream.open(fruPath,
-                       std::ios::in | std::ios::out | std::ios::binary);
-
-    vpdFileStream.seekg(startOffset, ios_base::cur);
-    vpdFileStream.read(reinterpret_cast<char*>(&completeVPDFile[0]), 65504);
-    completeVPDFile.resize(vpdFileStream.gcount());
-    vpdFileStream.clear(std::ios_base::eofbit);
+    vpdFileStream.exceptions(std::ifstream::badbit);
```
[the remark this answers is not in the document]
Comment (jinuthomas) [reply vpd_tool_impl.cpp]: added it , not sure if it is useful , so initially was reluctant to add it .

File: impl.hpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (GiridhariKrishnan) [reply impl.hpp:90]: Oh Okay, but I think it won't give errno description, if any like error().

**Review by GiridhariKrishnan — APPROVED** [review]

**Review by PriyangaRamasamy — APPROVED** [review]

**Review by SunnySrivastava1984 — APPROVED** [review]

**Review by GiridhariKrishna — APPROVED** [review]

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]
CI failing. Once it is passed I will approve.

**Comment by PriyangaRamasamy** [comment]
jenkins run tests please

**Review by Alpana07 — APPROVED** [review]
Code looks okay to me.

**Review by SunnySrivastava1984 — APPROVED** [review]

**Pull request merged**